sentry-ruby/lib/sentry/transport.rb

```diff
@@ -83,27 +92,85 @@ def generate_auth_header
 
     def encode(event)
       # Convert to hash
-      event_hash = event.to_hash
+      event_payload = event.to_hash
 
-      event_id = event_hash[:event_id] || event_hash["event_id"]
-      item_type = get_item_type(event_hash)
+      event_id = event_payload[:event_id] || event_payload["event_id"]
+      item_type = get_item_type(event_payload)
+
+      envelope_header = {
```

Comment:
nice refactor 👍

Reply:
I thought of this as a good intermediate intermediate step to having `Envelope` and `Item` classes later.